modules/core/src/main/scala/org/scalasteward/core/io/package.scala

```diff
@@ -21,21 +21,23 @@ import cats.implicits._
 import org.scalasteward.core.data.{GroupId, Update}
 
 package object io {
-  def isSourceFile(update: Update)(file: File): Boolean = {
+  def isSourceFile(update: Update, fileExtensions: Set[String])(file: File): Boolean = {
     val notInGitDir = !file.pathAsString.contains(".git/")
-    notInGitDir && isSpecificOrGenericSourceFile(update)(file)
+    notInGitDir && isSpecificOrGenericSourceFile(update, fileExtensions)(file)
   }
 
-  private def isSpecificOrGenericSourceFile(update: Update)(file: File): Boolean =
+  private def isSpecificOrGenericSourceFile(update: Update, fileExtensions: Set[String])(
+      file: File
+  ): Boolean =
     () match {
       case _ if isSbtUpdate(update)          => file.name === "build.properties"
       case _ if isScalafmtCoreUpdate(update) => file.name === ".scalafmt.conf"
-      case _                                 => isGenericSourceFile(file)
+      case _                                 => isGenericSourceFile(file, fileExtensions)
     }
 
-  private def isGenericSourceFile(file: File): Boolean = {
+  private def isGenericSourceFile(file: File, fileExtensions: Set[String]): Boolean = {
     val name = file.name
-    val allowedByExtension = file.extension.exists(Set(".scala", ".sbt", ".sc", ".yml"))
+    val allowedByExtension = file.extension.exists(fileExtensions)
     def allowedBySuffix: Boolean =
       Set(".sbt.shared").exists(suffix => name.endsWith(suffix) && !name.startsWith(suffix))
```

Comment:
It would be great if suffixes like `.sbt.shared` could be also handled by this feature. I guess we could just remove `val allowedByExtension` and use `fileExtensions` instead of `Set(".sbt.shared")` in `def allowedBySuffix`. WDYT?

Reply:
I agree, it makes sense. I merged these 2 tests.
Q: What should `!name.startsWith(suffix)` prevent ? That the suffix is the complete filename `name != suffix` ?

Reply:
That code was added in #1194 but it does not mention the reason for that check. But yes, I think I wanted to make sure that the suffix is a proper suffix.